sources/bottlerocket-variant/src/lib.rs

```diff
@@ -370,6 +381,20 @@ impl PartialEq<Variant> for &str {
     }
 }
 
+/// This can be used in a `build.rs` file to emit `cfg` values that can be used for conditional
+/// compilation based on the network backend. This function also emits rerun-if-changed so
+/// that packages sensitive to network backend will rebuild if the backend changes.
+/// TODO: Remove this function once systemd-networkd integration development is finished
+pub fn emit_net_backend_cfgs() {
+    let net_backend = if Some("1".to_string()) == std::env::var(NET_BACKEND_OVERRIDE_ENV).ok() {
+        NET_BACKEND_OVERRIDE
+    } else {
+        DEFAULT_NET_BACKEND
+    };
+    println!("cargo:rerun-if-env-changed={}", NET_BACKEND_OVERRIDE_ENV);
+    println!("cargo:rustc-cfg=net_backend=\"{}\"", net_backend);
+}
```

Comment:
I'd rather do this in a netdog-specific build script. It's not something most other crates will care about, while quite a few benefit from structured parsing of the variant tuple.

Reply:
> I'd rather do this in a netdog-specific build script. It's not something most other crates will care about, while quite a few benefit from structured parsing of the variant tuple.

Not sure why I didn't think that. I think when we discussed it @zmrow it was only netdog that needed this?

Reply:
That's fair.  My reasoning for putting it in the variant crate was "just in case we need it outside of netdog".  I'm fine with scoping it to netdog for now.